daphne_worker/src/dap.rs

```diff
@@ -547,14 +546,19 @@ impl<D> DapHelper<BearerToken> for DaphneWorkerConfig<D> {
             }
         }
 
-        // TODO Run requests in parallel.
-        let mut early_fails = HashMap::new();
+        let mut futures = Vec::new();
         for (durable_name, nonce_hex_set) in durable_requests.into_iter() {
-            let stub = namespace.id_from_name(&durable_name)?.get_stub()?;
-            let mut resp =
-                durable_post!(stub, DURABLE_REPORT_STORE_PUT_PROCESSED, &nonce_hex_set).await?;
+            futures.push(self.durable_post(
+                BINDING_DAP_REPORT_STORE,
+                DURABLE_REPORT_STORE_PUT_PROCESSED,
+                durable_name,
+                nonce_hex_set,
+            ));
+        }
 
-            let res: Vec<(String, TransitionFailure)> = resp.json().await?;
+        let mut early_fails = HashMap::new();
+        for future in futures.into_iter() {
```

Comment:
Does this serialize the order in which we process the futures? Say, for example, the first DO message takes forever to process, but the latter ones complete quickly. Will we block here waiting on a response from the first future? Is it possible instead to run this functionality in a callback that is invoked when the DO future returns an answer?

Reply:
> Does this serialize the order in which we process the futures? Say, for example, the first DO message takes forever to process, but the latter ones complete quickly. Will we block here waiting on a response from the first future?

I don't think any request is blocked on any other request. We `await` each request in the order in which they were issued, but waiting for one doesn't block handing of the other.

This is just my intuition, however. There may be hidden performance bottlenecks I don't see. 

> Is it possible instead to run this functionality in a callback that is invoked when the DO future returns an answer?
This can be done with `try_join_all()` from the `futures` crate. This is implemented in the next PR.